### PR TITLE
fix(ivy): ComponentFactory.create should clear host element content

### PR DESCRIPTION
**Pull request by AndrewKushnir**
Prior to this change, ComponentFactory.create function invocation in Ivy retained the content of the host element (in case host element reference or CSS selector is provided as an argument). This behavior is different in View Engine, where the content of the host element was cleared, except for the case when ShadowDom encapsulation is used (to make sure native slot projection works). This commit aligns Ivy and View Engine and makes sure the host element is cleared before component content insertion.

This PR resolves FW-1649.

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by AndrewKushnir** [comment]
Initial presubmit runs to verify the fix:
- [VE Presubmit](http://test/OCL:277548909:BASE:277549011:1572460362639:dc20e01d)
- [Ivy Presubmit](http://test/OCL:277548909:BASE:277549286:1572460428302:38540648)

**Comment by AndrewKushnir** [comment]
- [VE Presubmit](http://test/OCL:278034353:BASE:278034370:1572655880407:f406dd34)
- [Ivy Presubmit](http://test/OCL:278034353:BASE:278034434:1572655917648:eb64b35a)

**Comment by AndrewKushnir** [comment]
- [VE Presubmit](http://test/OCL:279517026:BASE:279517048:1573329137503:2e181072)
- [Ivy Presubmit](https://test.corp.google.com/ui#id=OCL:279517026:BASE:279517117:1573329177328:4ad45f95)

**Review by atscott — COMMENTED** [review]
LGTM. As discussed, please add a test case covering host elements with more than just a text node.

**Review by atscott — APPROVED** [review]

**Review by mhevery — APPROVED** [review]

**Comment by pkozlowski-opensource** [comment]
Initially I thought that this change limits our ability to fix #4946 but after digging into it more, I don't think it blocks anything. WE can go ahead with this PR from the existing GH issues point of view.

File: packages/core/src/render3/instructions/shared.ts [hunk 1]
```diff
@@ -678,32 +679,50 @@ function createViewBlueprint(bindingStartIndex: number, initialViewLength: numbe
...
   }
-  return rNode;
+
+  let rElement = typeof elementOrSelector === 'string' ?
+      renderer.querySelector(elementOrSelector) ! :
+      elementOrSelector;
+  ngDevMode && assertHostNodeExists(rElement, elementOrSelector);
+
+  // Always clear host element's content when Renderer3 is in use. For procedural renderer case we
+  // make it conditionally and depend on whether ShadowDom encapsulation is used (in which case the
```
Comment (kara) [inline packages/core/src/render3/instructions/shared.ts]:
```suggestion
  // make it conditional and depend on whether ShadowDom encapsulation is used (in which case the
```

**Review by kara — APPROVED** [review]
LGTM, once typo is fixed

**Pull request closed without merging**